lib/katello/tasks/pulp3_post_migration_check.rake

```diff
@@ -0,0 +1,38 @@
+require File.expand_path("../engine", File.dirname(__FILE__))
+
+namespace :katello do
+  desc "Runs a post Pulp3 migration check for supported content types."
+  task :pulp3_post_migration_check => :environment do
+    repository_types = Katello::Pulp3::Migration::REPOSITORY_TYPES
+
+    repository_types.each do |type|
+      filter = 'version_href is NULL OR remote_href is NULL'
+
+      unless type == Katello::Repository::DOCKER_TYPE
+        filter += ' OR publication_href is NULL'
+      end
+      repositories = Katello::Repository.with_type(type).where(filter)
+
+      if repositories.any?
+        $stderr.print("ERROR: at least one #{type} repository record has a NULL value for remote_href, version_href, publication_href\n")
+        exit 1
+      end
+
+      repositories = Katello::Repository.with_type(type)
```

Comment:
Is it necessary to filter out the repos where remote, version, or publication href are `nil`?  If they are `nil`, I think the execution wouldn't make it this far due to the `exit 1` on line 18.

Reply:
Fantastic catch!